test/test_main.py

```diff
@@ -33,3 +37,53 @@ def test_mixed(self):
         s = "0 1, 3 5-8, 9 12-10"
         parsed = parse_input_numbers(s)
         self.assertEqual(parsed, [0, 1, 3, 5, 6, 7, 8, 9, 10, 11, 12])
+
+
+class TestMongoDBIntegration(unittest.TestCase):
+    """
+    A few conditional integration test cases with MongoDB.
+
+    It can be configured to not fail the test suite when no MongoDB server
+    is running. In order to provide an instance easily, use Docker or Podman.
+
+    # MongoDB 4
+    docker run -it --rm --publish=27017:27017 mongo:4
+    """
+
+    HOST = "localhost"
+    PORT = 27017
+    DBNAME = "testdrive"
+    TIMEOUT_MS = 200
+
+    SKIP_IF_NOT_RUNNING = False
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = pymongo.MongoClient(
+            host=cls.HOST,
+            port=cls.PORT,
+            connectTimeoutMS=cls.TIMEOUT_MS,
+            serverSelectionTimeoutMS=cls.TIMEOUT_MS,
+        )
+        cls.db = cls.client.get_database(cls.DBNAME)
+        try:
+            cls.db.last_status()
+        except pymongo.errors.ServerSelectionTimeoutError:
+            if cls.SKIP_IF_NOT_RUNNING:
+                raise cls.skipTest(cls, reason="MongoDB server not running")
+            else:
+                raise
```

Comment:
Maybe there is some _Testcontainers for Python_ implementation for this? We can check on a subsequent iteration.